satpy/readers/seviri_l1b_native.py

```diff
@@ -366,58 +368,93 @@ def get_area_extent(self, dataset_id):
         # check if data is in Rapid Scanning Service mode (RSS)
         is_rapid_scan = self.trailer['15TRAILER']['ImageProductionStats']['ActualScanningSummary']['ReducedScan']
 
-        # The HRV channel in full disk mode comes in two separate areas, and each area has its own area extent stored
-        # in the trailer.
-        # In Rapid Scanning mode, only the "Lower" area (typically over Europe) is acquired and included in the files.
-        if (dataset_id['name'] == 'HRV') and (self.mda['is_full_disk'] or is_rapid_scan):
-
-            # get actual navigation parameters from trailer data
-            data15tr = self.trailer['15TRAILER']
-            HRV_bounds = data15tr['ImageProductionStats']['ActualL15CoverageHRV']
-
-            # lower window
-            lower_north_line = HRV_bounds['LowerNorthLineActual']
-            lower_west_column = HRV_bounds['LowerWestColumnActual']
-            lower_south_line = HRV_bounds['LowerSouthLineActual']
-            lower_east_column = HRV_bounds['LowerEastColumnActual']
-
-            lower_area_extent = self._calculate_area_extent(
-                center_point, lower_north_line, lower_east_column,
-                lower_south_line, lower_west_column, we_offset,
-                ns_offset, column_step, line_step
-            )
-
-            if is_rapid_scan:
-                return lower_area_extent
-
-            lower_nlines = lower_north_line - lower_south_line + 1
-            lower_ncols = lower_west_column - lower_east_column + 1
-
-            # upper window
-            upper_north_line = HRV_bounds['UpperNorthLineActual']
-            upper_west_column = HRV_bounds['UpperWestColumnActual']
-            upper_south_line = HRV_bounds['UpperSouthLineActual']
-            upper_east_column = HRV_bounds['UpperEastColumnActual']
-
-            upper_area_extent = self._calculate_area_extent(
-                center_point, upper_north_line, upper_east_column,
-                upper_south_line, upper_west_column, we_offset,
-                ns_offset, column_step, line_step
-            )
-
-            upper_nlines = upper_north_line - upper_south_line + 1
-            upper_ncols = upper_west_column - upper_east_column + 1
-
-            return [upper_area_extent, lower_area_extent, upper_nlines, upper_ncols, lower_nlines, lower_ncols]
-
-        # If the data was ordered in a defined ROI, the area extent is in one piece, the corner points are
-        # the same as for VISIR channels, and the HRV channel is having three times the amount of columns and rows.
+        # If we're dealing with HRV data, three different configurations of the data must be considered:
```

Comment:
Alright, so this function is waaay too long, more that 200 lines :) Not that I count lines, but too many things are happening here. So I can see that two methods could be extracted from here, one for HRV data, and another for the other channels. The HRV one could then be subdivided further between the FES/RSS case and the the ROI case. What do you think?

Reply:
Yeah, you're right! I'll fix this. Could I then create functions like 'get_area_extent_visir',  'get_area_extent_hrv' and 'get_area_extent_roi', just as an example? Maybe I could even split FES and RSS, I'll see when I get back to the coding.

Reply:
sounds good. I would suggest you prefix these with `_`, like `_get_area_extent_visir` which indicate private methods (as I don't think these should be part of the public API, right?)

Reply:
ok, that's probably the case. Should these methods, as well as the padding methods, still be part of the NativeMSGFileHandler class, or shall I put them separately at the end of the reader?

Reply:
i would think private methods within the file handler-class, unless there's a use for those methods in other classes (hrit)? in that case -> `seviri_base.py`

Reply:
Alright, thanks!